test/augmentation/test_augmentation.py

```diff
@@ -1623,3 +1634,94 @@ def test_gradcheck(self, device):
         }
         assert gradcheck(RandomMotionBlur(
             kernel_size=3, angle=(10, 30), direction=(-0.5, 0.5)), (inp, params), raise_exception=True)
+
+
+class TestRandomEqualize:
+    def smoke_test(self, device, dtype):
+        f = RandomEqualize(0.5)
+        repr = "RandomEqualize(p=0.5, return_transform=False)"
+        assert str(f) == repr
+
+    def test_random_equalize(self, device, dtype):
+        f = RandomEqualize(p=1.0, return_transform=True)
+        f1 = RandomEqualize(p=0., return_transform=True)
+        f2 = RandomEqualize(p=1.)
+        f3 = RandomEqualize(p=0.)
+
+        bs, channels, height, width = 1, 3, 20, 20
+
+        inputs = self.build_input(channels, height, width, device=device, dtype=dtype).squeeze(dim=0)
+        inputs.to(device)
+
+        row_expected = torch.tensor([
+            0.0000, 0.07843, 0.15686, 0.2353, 0.3137, 0.3922, 0.4706, 0.5490, 0.6275,
+            0.7059, 0.7843, 0.8627, 0.9412, 1.0000, 1.0000, 1.0000, 1.0000, 1.0000,
+            1.0000, 1.0000
+        ])
+        expected = self.build_input(channels, height, width, bs=1, row=row_expected,
+                                    device=device, dtype=dtype)
+        expected = expected.to(device)
+
+        identity = torch.eye(3, device=device)  # 3 x 3
+
+        assert_allclose(f(inputs)[0], expected)
+        assert_allclose(f(inputs)[1], identity)
+        assert_allclose(f1(inputs)[0], inputs)
+        assert_allclose(f1(inputs)[1], identity)
+        assert_allclose(f2(inputs), expected)
+        assert_allclose(f3(inputs), inputs)
+
+    def test_batch_random_equalize(self, device, dtype):
+        f = RandomEqualize(p=1.0, return_transform=True)
+        f1 = RandomEqualize(p=0., return_transform=True)
+        f2 = RandomEqualize(p=1.)
+        f3 = RandomEqualize(p=0.)
+
+        bs, channels, height, width = 2, 3, 20, 20
+
+        inputs = self.build_input(channels, height, width, bs, device=device, dtype=dtype)
+        inputs = inputs.to(device)
+
+        row_expected = torch.tensor([
+            0.0000, 0.07843, 0.15686, 0.2353, 0.3137, 0.3922, 0.4706, 0.5490, 0.6275,
+            0.7059, 0.7843, 0.8627, 0.9412, 1.0000, 1.0000, 1.0000, 1.0000, 1.0000,
+            1.0000, 1.0000
+        ])
```

Comment:
this need to be passed to device and dtype

Reply:
As we pass it to `build_input`, and the function is now returning `batch.to(device, dtype)`, I believe the row itself doesn't need to have `device` and `dtype` as well.

Reply:
true